roles/de-slapd/tasks/main.yml

```diff
@@ -1,54 +1,57 @@
 ---
-- include: prereqs.yml
+- name: Install packages
+  ansible.builtin.include_tasks: "{{ ansible_os_family }}.yml"
 
-- name: "Create the Database Configuration File"
-  become: true
-  command:
+- name: Check slapd is running
+  ansible.builtin.shell: >-
+    set -o pipefail;
+    ps -ef |
+    awk '/[s]lapd .*-F/{print $2}'
   args:
-    argv:
-      - "cp"
-      - "/usr/share/openldap-servers/DB_CONFIG.example"
-      - "/var/lib/ldap/DB_CONFIG"
+    executable: /bin/bash
+  register: _de_slap__isrunning
+  changed_when: false
+
+- name: Kill wrong slapd
+  ansible.builtin.command:
+    cmd: kill {{ _de_slap__isrunning.stdout }}
+  when:
+    - _de_slap__isrunning.stdout | length > 0
+  args:
+    removes: /proc/{{ _de_slap__isrunning.stdout }}/status
 
 - name: "Encode the Root Password"
-  become: true
-  command:
+  ansible.builtin.command:
   args:
     argv:
       - "slappasswd"
       - "-s"
       - "{{ root_password }}"
   register: slappasswd
+  failed_when: slappasswd.stdout is not regex('^\{SSHA\}.{32}$')
+  changed_when: false
+  tags:
+    - installation
 
+# TODO: change to new configuration-format
 - name: "Generate the OpenLDAP Configuration File"
-  become: true
-  template:
+  ansible.builtin.template:
     src: "slapd.conf.j2"
-    dest: "/etc/openldap/slapd.conf"
+    dest: "/etc/ldap/slapd.conf"
```

Comment:
is this directory name change is because of debian - or its not at all openldap you are trying to install?

Reply:
`/etc/ldap/` is the default config_dir in debians openldap-package.